### PR TITLE
Fix gamification PNG Error on NGINX

### PR DESCRIPTION
**Pull request by davidglezz**
FIX https://github.com/PrestaShop/PrestaShop/issues/17038

> issue with PS1.7.6.2 & PS1.7.5.2 & the gamification module v2.3.2.

> please add 1.7.6.3

Please test that now everything is fine with Apache and Nginx.

### CODE REVIEW

**Comment by davidglezz** [comment]
Random CI error?

**Comment by PierreRambaud** [comment]
Hi and thanks for your contribution.
We can't accept your pull request because if something is working on Apache, it should work on Nginx.
More, if you use a `/` you can't add your shop in a subdirectory, the `/` will refer to the root URL.

File: classes/Badge.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ class Badge extends ObjectModel
 
     public $awb;
     
-    const BAGDE_IMG_URL = '../modules/gamification/views/img/badges/';
+    const BAGDE_IMG_URL = _MODULE_DIR_ . 'gamification/views/img/badges/';
```
Comment (PierreRambaud) [inline classes/Badge.php]: _MODULE_DIR_ is a unix path, not an URL, it can't be used here.

Comment (davidglezz) [reply classes/Badge.php]: https://github.com/PrestaShop/PrestaShop/blob/1fcd73fef2befe84005596210a92a68222ef53dd/config/defines_uri.inc.php#L65

```php
define('_MODULE_DIR_', __PS_BASE_URI__.'modules/');
```
URI + 'modules/'

Comment (PierreRambaud) [reply classes/Badge.php]: My bad, I was thinking on `_PS_MODULE_DIR_` :man_facepalming:

**Comment by khouloudbelguith** [comment]
Hi @davidglezz,

In the BO => Shop Parameters => Merchant Expertise page, all the images are not displayed anymore. (checked with Nginx server and apache server).
**With PR:**
image
**Without PR:**
image

Thanks to check it.

**Comment by sarahdib** [comment]
Hello @davidglezz 
Thanks for this contribution, but since we found an error with this pull request we can't validate it.

Feel free to re-open this pull request when you have time to correct the error.

**Comment by davidglezz** [comment]
Hello!!

I request that the PR be reopened.

I have verified that it works. (today with 1.7.8.0)
After:
image


My arguments in favor of the PR
- Do not use relative path
- `_MODULE_DIR_` are generated by ps with user config
    - If not subdir:  `/modules/gamification...`
    - if subdir: `/subdir/modules/gamif...` because `_MODULE_DIR_ = '/subdir/modules/'`
- `_MODULE_DIR_` are used in a lot of modules,
If it does not work in this module, it should work in the other modules that use it.

**Comment by PierreRambaud** [comment]
@davidglezz I agree with the usage for `_MODULE_DIR` which is more relevant

**Comment by davidglezz** [comment]
Hi. you have forgotten to change the label to "waiting for QA" or the one that corresponds.

**Comment by khouloudbelguith** [comment]
Hi @davidglezz,

I tried with 1767 on Nginx server => **ok**
https://www.awesomescreenshot.com/video/799046?key=6b1dffb6d5884596fe45b19340db7374
But, I tried with the PR on 177x branch => **NOK**
image
I deleted the cache manually, I cleared the cache from the Performance page and I used an incognito browser => Same issue.
Thanks to check and feedback.

**Comment by davidglezz** [comment]
@khouloudbelguith This should not have different behaviors with different versions of Prestashop.

Just use a constant that exists since 1.5, 1.4 or earlier.

Did you check the generated url? as I show in a screenshot https://github.com/PrestaShop/gamification/pull/72#issuecomment-660522885

did you try to access the url of the image?

**Review by rblaurin — APPROVED** [review]

**Comment by ngodefroy** [comment]
Hello @davidglezz,

TEST OK on 177x
https://drive.google.com/file/d/1vuZv2xEjOMkve6uQ_KKd6ih6T8wo_xU9/view

Thank you

**Comment by atomiix** [comment]
Thanks @davidglezz !

**Pull request merged**